java9/src/main/java/net/nycjava/java9/api/J9Process.java

```diff
@@ -4,21 +4,23 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 
-import static java.lang.Thread.sleep;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 public class J9Process {
+
     public static void main(String[] args) throws IOException, InterruptedException {
-        Process process = new ProcessBuilder("ls", "-alR")
+        // Print all java files. find ~/ to see timeout in play
+        Process process = new ProcessBuilder("find", ".", "-name", "*.java")
                 .start();
+
+        System.out.println(String.format("PID: %s", process.pid()));
+
         process.onExit()
-                .orTimeout(1000, MILLISECONDS)
+                .orTimeout(100, MILLISECONDS)
                 .thenAccept(p -> new BufferedReader(new InputStreamReader(p.getInputStream()))
                         .lines()
-                        .filter(s -> s.startsWith("total"))
                         .forEach(System.out::println));
-        process.waitFor();
-        sleep(100);
-        System.out.println("Done");
+
+        process.waitFor(); // wait till process has terminated
```

Comment:
I trust you discovered the sleep(100) isn't required? I thought it was for the onExit to run

Reply:
Not for the find example we have